woocommerce/woocommerce-paypal-payments
# Create the pay later tab

**Issue**: #961

---

### Description

Creates the new Pay Later tab to host settings for Pay Later messaging and also the Pay Later button.
The new Pay Later tab will include:

- All existing Pay Later Messaging settings. (with improved design).
- New settings to manage Pay Later button.
- The previews.

<img width="1323" alt="Screen Shot 2022-11-02 at 19 08 51" src="https://user-images.githubusercontent.com/11319597/199526598-da7c94f0-9ec7-4cf8-a6b9-b4eb0ae45118.png">

The tab will be displayed only when the country is eligible for Pay Later.
Also the PR includes the migration script from old settings, which will update the values in database for the new settings on plugin update just once.

### Steps to Test




1. Check the new Pay Later tab with all settings.
2. Check if conditionally display of fields works.
3. If Vault is enabled then the Pay Later button and messaging are disabled with disabled checkbox and message.
4. Pay Later tab is not displayed when the store country is not eligible for Pay Later.
5. The Pay Later button is removed/added in Pay Later button preview regarding of the Pay Later Button enabled/disabled checkbox.
6. The messaging previews are working for each location regarding to appropriate settings.
7. When plugin is upgraded the old settings are migrated to the new Pay Later tab.
8. The migration works only once.

---

Closes #961 .
